src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,29 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, CompletableFuture<UserData>> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return cache.computeIfAbsent(userId, key ->
+                CompletableFuture.supplyAsync(() -> {
+                    try {
+                        TimeUnit.MILLISECONDS.sleep(200);
+                    } catch (InterruptedException e) {
+                        throw new RuntimeException(e);
+                    }
+                    return new UserData(userId, "Details for " + userId);
```

Comment:
let's move this sleep and `UserData` object build logic to a separate method and call it in `computeIfAbsent()` via a method reference